Summarycheckout

desafree opened the pull request:
Add summary checkout detail. Checkout functionality still missing

desafree committed:
feat:create checkout form page

-create form
-integrate form style
-integrate validation

desafree committed:
feat: create summary form

desafree committed:
refactor:improve clarity and reausability

-replace unecessary style with stack component
-add type for items
-remove unnecessary styled component in itempreviewsummary
-replace boilerplate of items
-add internal link functionality to button component

desafree committed:
refactor: improve readibility in summary form

desafree committed:
refactor: replace flex style with stack component

desafree committed:
refactor: improve readability

-remove unnecessary radio styled component
-add alt value in itempreviewsummary component

rams23 commented on src/checkout/components/Totals/Totals.tsx:
```diff
@@ -0,0 +1,37 @@
+        title={translate("checkout.summary.grand-total")}
+        value={calculateGrandTotal(total)}
+      ></TotalRow>
+    </Stack>
```
I would make shipping and vat props, since the logic could change, and maybe the value can come from an api, so i think it's better to calculate it into an hook and pass them as props

rams23 commented on src/_shared/components/Button/Button.tsx:
```diff
@@ -16,13 +19,39 @@ const Button: FC<Props> = ({
   variant = "primary",
   href,
   onClick,
+  type,
+  internalLink,
 }) => {
+  if (internalLink) {
+    return (
+      <ButtonMUI
+        disabled={disabled}
+        variant={variant}
+        component={Link}
+        to={internalLink}
```
I'm not sure that mixing button and link is a good idea, can be misleading

rams23 commented on src/checkout/components/Totals/Totals.tsx:
```diff
@@ -0,0 +1,37 @@
+  const translate = useTranslate();
+
+  return (
+    <Stack direction="column" spacing="8px">
```
mui has a spacing system that uses numbers as identifier of theme dimensions: for ui consistency it is better to use them instead of raw pixels, so it can be controlled directly from the theme. 8 is 1, 16 is 2 etc

desafree committed:
refactor: improve clarity and modularity

-passing vat, shipping and grandtotal as props in totals component
-replace button with link for internal linking
-improve spacing value consistency for grid and stack components

rams23 commented on src/checkout/components/Summary/Summary.tsx:
```diff
@@ -6,23 +6,29 @@ import {
   const translate = useTranslate();
   const items = useSelectProductsFromCart();
-  const total = calculateTotalFromItems(items);
+  const { total, shipping, vat, grandTotal } = calculateTotals(items);
```
calculateVat and Shipping are not used anymore i think

desafree committed:
refactor: remove calculatetotal and calculatevat
